examples/green_screen/MultiDeviceCapturer.h

```diff
@@ -0,0 +1,220 @@
+constexpr std::chrono::microseconds MAX_ALLOWABLE_TIME_OFFSET_ERROR_FOR_IMAGE_TIMESTAMP(50);
+
+class MultiDeviceCapturer
+{
+public:
+    // Set up all the devices. Note that the index order isn't necessarily preserved, because we might swap with master
+    MultiDeviceCapturer(const vector<int> &device_indices, int32_t color_exposure_usec, int32_t powerline_freq)
+    {
+        bool master_found = false;
+        if (device_indices.size() == 0)
+        {
+            throw std::runtime_error("Capturer must be passed at least one camera!");
+        }
+        for (int i : device_indices)
+        {
+            devices.emplace_back(k4a::device::open(i)); // construct a device using this index
+            // If you want to synchronize cameras, you need to manually set both their exposures
+            devices.back().set_color_control(K4A_COLOR_CONTROL_EXPOSURE_TIME_ABSOLUTE,
+                                             K4A_COLOR_CONTROL_MODE_MANUAL,
+                                             color_exposure_usec);
+            // This setting compensates for the flicker of lights due to the frequency of AC power in your region. If
+            // you are in an area with 50 Hz power, this may need to be updated (check the docs for
+            // k4a_color_control_command_t)
+            devices.back().set_color_control(K4A_COLOR_CONTROL_POWERLINE_FREQUENCY,
+                                             K4A_COLOR_CONTROL_MODE_MANUAL,
+                                             powerline_freq);
+            // We treat the first device found with a sync out cable attached as the master. If it's not supposed to be,
+            // unplug the cable from it. Also, if there's only one device, just use it
+            if ((devices.back().is_sync_out_connected() && !master_found) || device_indices.size() == 1)
+            {
+                // If this isn't already the first and only device, make it the first
+                if (devices.size() > 1)
+                {
+                    std::swap(devices.front(), devices.back());
+                }
+                master_found = true;
+            }
+            else if (!devices.back().is_sync_in_connected() && !devices.back().is_sync_out_connected())
+            {
+                throw std::runtime_error("Each device must have sync in or sync out connected!");
+            }
+            else if (!devices.back().is_sync_in_connected())
+            {
+                throw std::runtime_error("Non-master camera found that doesn't have the sync in port connected!");
+            }
+        }
+        if (!master_found)
+        {
+            throw std::runtime_error("No device with sync out connected found!");
+        }
+    }
+
+    // configs[0] should be the master, the rest subordinate
+    void start_devices(const k4a_device_configuration_t &master_config,
+                       const std::vector<k4a_device_configuration_t> &sub_configs)
```

Comment:
Why is there a concept of multiple sub_configs. Can we optimize down to a single sub_config that will be applied to all subordinates?